search/src/main/kotlin/com/ritense/search/autodeployment/SearchListColumnDefinitionDeploymentService.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2015-2024 Ritense BV, the Netherlands.
+ *
+ * Licensed under EUPL, Version 1.2 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://joinup.ec.europa.eu/collection/eupl/eupl-text-eupl-12
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.ritense.search.autodeployment
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import com.ritense.search.domain.SearchListColumn
+import com.ritense.search.domain.SearchListColumnConfigurationAutoDeploymentFinishedEvent
+import com.ritense.search.service.SearchListColumnService
+import mu.KotlinLogging
+import org.springframework.boot.context.event.ApplicationReadyEvent
+import org.springframework.context.ApplicationEventPublisher
+import org.springframework.context.event.EventListener
+import org.springframework.core.Ordered
+import org.springframework.core.annotation.Order
+import org.springframework.core.io.Resource
+import org.springframework.core.io.ResourceLoader
+import org.springframework.core.io.support.ResourcePatternUtils
+import java.io.IOException
+
+class SearchListColumnDefinitionDeploymentService(
```

Comment:
For deployment of search fields we already have something. There's a `SearchFieldDeployer` in the `search` module that is meant to deploy search fields for different things. To support a new related type, a subclass can be made of that. See the `TaskSearchFieldDeployer` in the `process-document` module as an example.

Reply:
Yeah that deploys search fields not searchListColumns